api/src/main/java/org/openmrs/validator/DiagnosisValidator.java

```diff
@@ -0,0 +1,86 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.validator;
+
+import org.openmrs.CodedOrFreeText;
+import org.openmrs.ConditionVerificationStatus;
+import org.openmrs.Diagnosis;
+import org.openmrs.Encounter;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.ValidationUtils;
+import org.springframework.validation.Validator;
+
+/**
+ * Validates {@link Validator} objects
+ * This class ensures that the condition object is valid and properly structured
+ * 
+ * @since 2.2
+ */
+@Handler(supports = {Validator.class}, order = 50)
+public class DiagnosisValidator implements Validator {
+
+	/**
+	 * @should support Diagnosis class
+	 */
+	@Override
+	public boolean supports(Class<?> aClass) {
+		return Diagnosis.class.isAssignableFrom(aClass);
+	}
+
+	/**
+	 * @see org.springframework.validation.Validator#validate(java.lang.Object,
+	 *      org.springframework.validation.Errors)
+	 * @should fail validation if rank is null
+	 * @should fail validation if certainty is null
+	 * @should fail validation if diagnosis is null
+	 * @should fail validation if diagnosisId is null
+	 * @should fail for a null object
+	 */
+	@Override
+	public void validate(Object o, Errors errors) {
+		Diagnosis diagnosis = (Diagnosis)o;
+		
+		if(diagnosis == null){
+			throw new APIException("Diagnosis can't be null");
+		} else if (diagnosis.getVoided()) {
+			return;
+		}
+		
+		//Ensure there is an encounter associated with a diagnosis
```

Comment:
You can remove these comments because they describe what is obvious https://improvingsoftware.com/2011/06/27/5-best-practices-for-commenting-your-code/